Failed to build package hiredis ssl on windows*

**Failed to build package hiredis ssl on windows 64 bit with VS 2017/2019**

To Reproduce
Steps to reproduce the behavior:
vcpkg upgrade hiredis[ssl]:x64-windows-static
vcpkg upgrade hiredis[ssl]:x64-windows-static --no-dry-run
vcpkg install hiredis[ssl]:x64-windows-static --recurse
whereas vcpkg install hiredis:x64-windows-static is building successfully
Failure logs
\vcpkg\buildtrees\hiredis\src\4889033d94-d35835c897.clean\ssl.c(37): fatal error C1083: Cannot open include file: 'pthread.h': No such file or directory

--- a/ssl.c
+++ b/ssl.c
@@ -36,7 +36,7 @@
 #include <assert.h>
 #include <errno.h>
 #include <string.h>
-#ifdef WIN32
+#ifdef _WIN32
 #include <windows.h>
 #else
 #include <pthread.h>
@@ -95,7 +95,7 @@ redisContextFuncs redisContextSSLFuncs;
 #endif
 
 #ifdef HIREDIS_USE_CRYPTO_LOCKS
-#ifdef WIN32
+#ifdef _WIN32
 typedef CRITICAL_SECTION sslLockType;
 static void sslLockInit(sslLockType* l) {
     InitializeCriticalSection(l);

